commit 880f81914232c939e7a83cce9365a90ee9e9600f
Author: Nikita Nazarenko <nnazarenko@radiofid.ru>
Date:   Tue Mar 3 16:51:47 2015 +0300

    usb: option: interpret error -71 as debug, not an error

--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -2030,7 +2030,7 @@ static void option_instat_callback(struc
 			dev_dbg(dev, "%s: type %x req %x\n", __func__,
 				req_pkt->bRequestType, req_pkt->bRequest);
 		}
-	} else if (status == -ENOENT || status == -ESHUTDOWN) {
+	} else if (status == -ENOENT || status == -ESHUTDOWN || status == -71) {
 		dev_dbg(dev, "%s: urb stopped: %d\n", __func__, status);
 	} else
 		dev_err(dev, "%s: error %d\n", __func__, status);
