--- a/drivers/gpio/Makefile
+++ b/drivers/gpio/Makefile
@@ -35,6 +35,7 @@
 obj-$(CONFIG_GPIO_ICH)		+= gpio-ich.o
 obj-$(CONFIG_GPIO_IOP)		+= gpio-iop.o
 obj-$(CONFIG_GPIO_LV)		+= gpio-lv.o
+obj-$(CONFIG_GPIO_TR)		+= gpio-tr.o
 obj-$(CONFIG_GPIO_IT8761E)	+= gpio-it8761e.o
 obj-$(CONFIG_GPIO_JANZ_TTL)	+= gpio-janz-ttl.o
 obj-$(CONFIG_GPIO_KEMPLD)	+= gpio-kempld.o

--- a/drivers/gpio/Kconfig	2016-11-15 16:35:02.564232313 +0300
+++ b/drivers/gpio/Kconfig	2016-11-15 16:35:31.838661034 +0300
@@ -924,4 +924,9 @@
         bool 
         prompt "Levelshifter gpio wrapper"
         default n
+
+config GPIO_TR
+        bool 
+        prompt "R2 gpio wrapper"
+        default n
 endif
