commit d5fab26e19363e5f539b7d7536bbbed476de4b9b
Author: Nikita Nazarenko <nnazarenko@radiofid.ru>
Date:   Tue Mar 3 16:49:47 2015 +0300

    usb: option: add WeTelecom WPD-200 module

--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -545,6 +545,7 @@ enum option_blacklist_reason {
 #define WETELECOM_PRODUCT_6802			0x6802
 #define WETELECOM_PRODUCT_WMD300		0x6803
 #define WETELECOM_WPD_600			0x9070
+#define WETELECOM_WPD_200			0x6801
 
 struct option_blacklist_info {
 	/* bitfield of interface numbers for OPTION_BLACKLIST_SENDSETUP */
@@ -2016,6 +2017,7 @@ static const struct usb_device_id option
 	{ USB_DEVICE(VIATELECOM_VENDOR_ID, VIATELECOM_PRODUCT_CDS7) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(WETELECOM_VENDOR_ID, WETELECOM_PRODUCT_WMD200, 0xff, 0xff, 0xff) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(WETELECOM_VENDOR_ID, WETELECOM_PRODUCT_WPD600, 0xff, 0xff, 0xff) },
+	{ USB_DEVICE_AND_INTERFACE_INFO(WETELECOM_VENDOR_ID, WETELECOM_PRODUCT_WPD200, 0xff, 0xff, 0xff) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(WETELECOM_VENDOR_ID, WETELECOM_PRODUCT_6802, 0xff, 0xff, 0xff) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(WETELECOM_VENDOR_ID, WETELECOM_PRODUCT_WMD300, 0xff, 0xff, 0xff) },
 	{ } /* Terminating entry */
